### PR TITLE
Extensions and Orderable

### PR DESCRIPTION
**Pull request by sfnelson**

### CODE REVIEW

File: app/components/concerns/katalyst/tables/orderable.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      def orderable
+        @orderable
+      end
+
+      def with_orderable(url:,
+                         id: "#{self.id}-orderable-form",
+                         scope: "order[#{collection.items.model_name.plural}]")
+        header_row_component.include(HeaderRow)
+        body_row_component.include(BodyRow)
+        @orderable = FormComponent.new(table: self, url: url, id: id, scope: scope)
```
Comment (mattr) [inline app/components/concerns/katalyst/tables/orderable.rb:33]: (opt) if we return `self` here, could we make this chainable? (Is that something we want to support?)
```ruby
table = TableComponent.new(collection:).with_orderable(url:)
```

File: app/components/concerns/katalyst/tables/orderable.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      class FormComponent < ViewComponent::Base # :nodoc:
+        attr_reader :id, :url, :scope
+
+        def initialize(table:, id:, url:, scope:)
+          super
+
+          @table = table
+          @id = id
+          @url = url
+          @scope = scope
```
Comment (mattr) [inline app/components/concerns/katalyst/tables/orderable.rb:85]: Has our alignment cop changed? Seeing quite a few instances of this.

**Review by mattr — APPROVED** [review]
LGTM.

I think it would be good to follow up with some examples of how to more generally build and add custom extensions for a table.

**Review by AlanCornthwaiteKatalyst — APPROVED** [review]

File: app/components/concerns/katalyst/tables/orderable.rb (hunk 1 above)
Comment (sfnelson) [reply app/components/concerns/katalyst/tables/orderable.rb:33]: Good idea

File: app/components/concerns/katalyst/tables/orderable.rb (hunk 2 above)
Comment (sfnelson) [reply app/components/concerns/katalyst/tables/orderable.rb:85]: Yeah I'm really confused why it's not doing anything

Comment (sfnelson) [reply app/components/concerns/katalyst/tables/orderable.rb:85]: Fixed 👍

**Pull request merged**